opt: TestStats fails with TAGS=fast_int_set_large

```
make test TAGS=fast_int_set_large PKG=./pkg/sql/opt/memo

--- FAIL: TestStats (0.11s)
    --- FAIL: TestStats/scan (0.02s)
        memo_test.go:305: output didn't match expected:
            @@ -2,13 +2,13 @@
              ├── columns: x:1(int!null) y:2(int) z:3(int) s:4(string)
              ├── stats: [rows=167.000668, distinct(2)=2, null(2)=167.000668]
              ├── key: (1)
              ├── fd: (1)-->(2-4)
              └── scan t@secondary
            -      ├── columns: x:1(int!null) y:2(int)
            +      ├── columns: x:1(int!null) y:2(int!null)
                   ├── constraint: /2/1
                   │    ├── [/NULL/5 - /NULL/5]
                   │    └── [/5 - /5]
            -      ├── stats: [rows=1.95200401, distinct(1)=1.95200401, null(1)=0, distinct(2)=1.95200401, null(2)=1.95200401]
            +      ├── stats: [rows=1.95200401, distinct(1)=1.95200401, null(1)=0, distinct(2)=1.95200401, null(2)=0]
                   ├── key: (1)
                   └── fd: (1)-->(2)
```

--- a/pkg/sql/opt/memo/statistics_builder.go
+++ b/pkg/sql/opt/memo/statistics_builder.go
@@ -640,7 +640,7 @@ func (sb *statisticsBuilder) buildScan(scan *ScanExpr, relProps *props.Relationa
 	// unconstrained scan over a partial index. The selectivity of the partial
 	// index predicate expression must be applied to the underlying table stats.
 	if scan.Constraint == nil && scan.InvertedConstraint == nil {
-		notNullCols := relProps.NotNullCols
+		notNullCols := relProps.NotNullCols.Copy()
 		// Add any not-null columns from the predicate constraints.
 		for i := range pred {
 			if c := pred[i].ScalarProps().Constraints; c != nil {
@@ -761,7 +761,7 @@ func (sb *statisticsBuilder) constrainScan(
 
 	// Set null counts to 0 for non-nullable columns
 	// ---------------------------------------------
-	notNullCols := relProps.NotNullCols
+	notNullCols := relProps.NotNullCols.Copy()
 	if constraint != nil {
 		// Add any not-null columns from this constraint.
 		notNullCols.UnionWith(constraint.ExtractNotNullCols(sb.evalCtx))

